### PR TITLE
synchronize servlet request for `BaseMetricsRESTProxyServlet`

### LINKED ISSUE #24590: Test Failure: io.openliberty.microprofile.metrics.internal.monitor_fat.TestEnableDisableFeaturesTest.testEDF5
RTC 294706

```
testEDF5:java.lang.Throwable: 2023-02-11-10:21:17:074 There were 2 errors:
  java.io.IOException(Server returned HTTP response code: 500 for URL: https://localhost:8020/metrics?scope=vendor)
  java.lang.Exception(Errors/warnings were found in server TestEDF5 logs:
<br>[2/11/23, 10:21:11:341 PST] 0000002c com.ibm.ws.webcontainer.util.ApplicationErrorUtils           E SRVE0777E: Exception thrown by application class 'com.ibm.ws.session.http.AbstractHttpSession.isNew:218'
<br>[2/11/23, 10:21:11:572 PST] 0000002c com.ibm.ws.webcontainer.webapp                               E SRVE0315E: An exception occurred: java.lang.Throwable: java.lang.IllegalStateException: The method is called on an invalidated session: 58W_3izBZhmfEjRlEGi-Qg2)
at componenttest.custom.junit.runner.FATRunner.newThrowableWithTimeStamp(FATRunner.java:338)
at componenttest.custom.junit.runner.FATRunner.access$700(FATRunner.java:70)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:304)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:364)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:178)
  java.io.IOException(Server returned HTTP response code: 500 for URL: https://localhost:8020/metrics?scope=vendor)
  java.lang.Exception(Errors/warnings were found in server TestEDF5 logs:
<br>[2/11/23, 10:21:11:341 PST] 0000002c com.ibm.ws.webcontainer.util.ApplicationErrorUtils           E SRVE0777E: Exception thrown by application class 'com.ibm.ws.session.http.AbstractHttpSession.isNew:218'
<br>[2/11/23, 10:21:11:572 PST] 0000002c com.ibm.ws.webcontainer.webapp                               E SRVE0315E: An exception occurred: java.lang.Throwable: java.lang.IllegalStateException: The method is called on an invalidated session: 58W_3izBZhmfEjRlEGi-Qg2)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:204)
```

### PR DESCRIPTION
**Pull request by Channyboy**
#build
fixes #24590

### CODE REVIEW

**Comment by Channyboy** [comment]
Synchronize requests for the `/metrics` servlet.

It could be that multiple requests are made where both requests may use the same session instance. 
However this can lead to one request ending and invalidating the session for the other.
Specifically if the session is ended by the first request before the second request *retrieves* the session.

**Comment by Channyboy** [comment]
Failures in the build are unrelated to the changes in this PR.
However, will rebuild as one child build did not complete.

**Comment by Channyboy** [comment]
#build

File: dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/BaseMetricsRESTProxyServlet.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public abstract class BaseMetricsRESTProxyServlet extends HttpServlet {
 
     /** {@inheritDoc} */
     @Override
-    protected void service(final HttpServletRequest request, final HttpServletResponse response) throws ServletException, IOException {
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/BaseMetricsRESTProxyServlet.java:38]: I am not sure added `synchronized` will help here as `getAndSetRESTHandlerContainer` and `handleWithDelegate` are already `synchronized`.  Do you know why the session is invalidated? Did the session got invalidated in the code explicitly or due to timeout?

**Pull request closed without merging**